app/controllers/events_controller.rb

```diff
@@ -53,23 +53,15 @@ def destroy
 
   private
 
-  def set_event
-    @event = Event.find_by_id(params[:id]) || raise_not_found
-  end
-
   def set_project
     @project = Project.find_by_id(params[:project_id]) || raise_not_found
   end
 
-  def event_params
-    params.require(:event).permit(:title, :description, :date)
-  end
-
-  def authorize_project_access
-    authorize @project, :access?
+  def set_event
+    @event = @project.events.find_by_id(params[:id]) || raise_not_found
```

Comment:
Thanks, I just found out that Rails automatically rescues the ActiveRecord::RecordNotFound exception by rendering the 404 error page. :)